pkg/controller/deployment/deployment_controller.go

```diff
@@ -1002,11 +1107,7 @@ func (dc *DeploymentController) reconcileOldReplicaSets(allRSs []*extensions.Rep
 	if err != nil {
 		return false, fmt.Errorf("could not find available pods: %v", err)
 	}
-
-	_, maxUnavailable, err := deploymentutil.ResolveFenceposts(&deployment.Spec.Strategy.RollingUpdate.MaxSurge, &deployment.Spec.Strategy.RollingUpdate.MaxUnavailable, deployment.Spec.Replicas)
```

Comment:
Validation guarantees that we will never hit this error here:
https://github.com/kubernetes/kubernetes/blob/2524cadeeeb58fe282050dd1ef7bb7e85ff0099e/pkg/apis/extensions/validation/validation.go#L188

On Tue, May 24, 2016 at 11:36 PM, Phillip Wittrock <notifications@github.com
> wrote:

> In pkg/controller/deployment/deployment_controller.go
> <https://github.com/kubernetes/kubernetes/pull/20273#discussion_r64475797>
> :
>
> > @@ -1002,11 +1107,7 @@ func (dc *DeploymentController) reconcileOldReplicaSets(allRSs []*extensions.Rep
> >  	if err != nil {
> >  		return false, fmt.Errorf("could not find available pods: %v", err)
> >  	}
> > -
> > -	_, maxUnavailable, err := deploymentutil.ResolveFenceposts(&deployment.Spec.Strategy.RollingUpdate.MaxSurge, &deployment.Spec.Strategy.RollingUpdate.MaxUnavailable, deployment.Spec.Replicas)
>
> Why don't we return false in the event of an error anymore?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/20273/files/f0e6921fbe1628d2f462ebd69b23d865ea3839fe#r64475797>
>